x-pack/plugins/security_solution/public/detections/components/detection_engine_filters/detection_engine_filters.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useMemo, useState } from 'react';
+import type { FilterControlConfig } from '@kbn/alerts-ui-shared';
+import { AlertConsumers } from '@kbn/rule-data-utils';
+import { createKbnUrlStateStorage, Storage } from '@kbn/kibana-utils-plugin/public';
+import { ControlGroupRenderer } from '@kbn/controls-plugin/public';
+import type { AlertFilterControlsProps } from '@kbn/alerts-ui-shared/src/alert_filter_controls';
+import { AlertFilterControls } from '@kbn/alerts-ui-shared/src/alert_filter_controls';
+import { useHistory } from 'react-router-dom';
+import { useKibana } from '../../../common/lib/kibana';
+import { DEFAULT_DETECTION_PAGE_FILTERS } from '../../../../common/constants';
+import { URL_PARAM_KEY } from '../../../common/hooks/use_url_state';
+import { useSpaceId } from '../../../common/hooks/use_space_id';
+import type { SecuritySolutionDataViewBase } from '../../../common/types';
+import { SECURITY_ALERT_DATA_VIEW } from '../../constants';
+
+export type DetectionEngineFiltersProps = Pick<
+  AlertFilterControlsProps,
+  'filters' | 'onFiltersChange' | 'query' | 'timeRange' | 'onInit'
+> & {
+  indexPattern?: SecuritySolutionDataViewBase;
+};
+
+export const DetectionEngineFilters = ({ indexPattern, ...props }: DetectionEngineFiltersProps) => {
```

Comment:
Consider adding couple unit tests for this file

Reply:
Addressed in 7076779f3d55b5ee6673644b562a2185ddb52ef8